### ISSUE TITLE
RGP fasta 

### ISSUE THREAD

**Issue by SamVaGo** [labgem/PPanGGOLiN#163]
Dear

I have a large set of genomes of which I wanted to analyze the RGPs using your pipeline. I have run the commandline, extracted the fasta file from the RGPs, which gives me a files with contigs named after the organism ID and the CDS. Afterwards, I ran prokka to annotate this fasta file. Is there any way to link my annotated genes to an RGP? I cant seem to link any of the RGP / spots output files (summary or spots overview) to the CDS from the RGP fasta file.

Basically, I want to know on which spot or RGP a certain annotation is located.

Many thanks for your help!

Best
Sam

**Comment by dvallenet** [comment]
Dear Sam,

The best way to do this is probably to run ppanggolin directly from the gff files obtained with proka on the complete genomes with the command : ppanggolin workflow --anno ORGANISMS_ANNOTATION_LIST 
If I've understood your problem correctly?

Doc here:
https://github.com/labgem/PPanGGOLiN/wiki/Basic-usage-and-practical-information
https://github.com/labgem/PPanGGOLiN/wiki/PPanGGOLiN---step-by-step-pangenome-analysis#annotation


Best,

David

**Comment by SamVaGo** [comment]
Dear David

Thank you for your response. I was afraid so.
From my current workflow, I did extra analysis on the fasta file that was written from the RGPs (using ppanggolin fasta --rgp). But in this analysis, all results are linked to the contig name of the resulting fasta file, which is 'organismID + CDS', that i can't link back to the actual RGP or spot.

If there is no other option, I will rerun the analysis on the gff files.

Many thanks for your help!

Best
Sam

**Comment by dvallenet** [comment]
Sorry, but I can't think of a simpler solution.

**Comment by axbazin** [comment]
Hi,

I don't know if it can be a solution, but when I want to look for RGP or spots that have particular genes or functions, I use a reference protein sequence of said gene and seach it in the pangenome using `ppanggolin align`.
You can use it as such:

`ppanggolin align -p pangenome.h5 --proteins mygene.faa -o align_on_mygene --getinfo`

And in one of the resulting file, you shoud have `info_input_seq.tsv` which should list the RGP and spots in which the gene is found.

Adelme

**Comment by SamVaGo** [comment]
Thank you Adelme for this possible solution!
Eventually, I found a reference to the CDS in the gexf file of every spot. I combined all these files of all identified spots  (>200 files) and extracted from each file the edges (eg. spot_0.gexf$edges), added to each identified CDS the spot (according to the filename) and combined all rows. In that way my resulting dataset contained a column with the (specimenID +) CDS and a column in which spot it was found. Hopefully no erroneous result using these steps :).

Many thanks!

Best
Sam
